sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultSettingValue.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Azure.Security.KeyVault.Administration
+{
+    /// <summary>
+    /// An account setting value.
+    /// </summary>
+    public class KeyVaultSettingValue
+    {
+        internal KeyVaultSettingValue(string value, KeyVaultSettingType? settingType)
+        {
+            // Service may not response with settingType for boolean (originally only supported value type).
+            if (settingType == null || settingType == KeyVaultSettingType.Boolean)
+            {
+                if (!bool.TryParse(value, out bool result))
+                {
+                    throw InvalidSettingType(KeyVaultSettingType.Boolean);
+                }
+
+                ValueBool = result;
+                SettingType = KeyVaultSettingType.Boolean;
+            }
+            else
+            {
+                ValueString = value;
+                SettingType = settingType;
+            }
+        }
+
+        internal KeyVaultSettingValue(bool value)
+        {
+            ValueBool = value;
+            SettingType = KeyVaultSettingType.Boolean;
+        }
+
+        /// <summary>
+        /// Gets the type specifier of the value.
+        /// </summary>
+        internal KeyVaultSettingType? SettingType { get; }
+
+        /// <summary>
+        /// Gets the boolean value of this account setting if <see cref="Type"/> is <see cref="KeyVaultSettingType.Boolean"/>.
+        /// </summary>
+        /// <returns>A boolean value if <see cref="Type"/> is <see cref="KeyVaultSettingType.Boolean"/>.</returns>
+        /// <exception cref="InvalidOperationException">The <see cref="Type"/> is not <see cref="KeyVaultSettingType.Boolean"/>, or the value cannot be normalized as a Boolean.</exception>
+        public bool AsBoolean() => CheckType(KeyVaultSettingType.Boolean, ValueBool);
+
+        /// <summary>
+        /// Gets the raw string value of this account setting.
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            string conversionMethod = null;
+            if (SettingType == KeyVaultSettingType.Boolean)
+            {
+                conversionMethod = nameof(AsBoolean);
+            }
+
+            return conversionMethod == null
+                ? $"{nameof(KeyVaultSetting)}: {nameof(SettingType)}={SettingType}"
+                : $"{nameof(KeyVaultSetting)}: {nameof(SettingType)}={SettingType}, {conversionMethod}()=>Value";
```

Comment:
What do these look like?  Do we want the result to start with "KeyValueSetting: " when this is `KeyVaultSettingValue`?  Was the `=>Value` missing the actual value?

Reply:
I copied this from FR, which hardcodes `=>Value` hence why I called out I found this odd over Teams. Personally, I'd rather `KeyVaultSetting.ToString` return `{Name}: {Value}` and `KeyVaultSettingValue.ToString` return just `{Value}` and get rid of `content`; though, I'd still need to figure out what to call it in the model factory since I can't use `Value` with an automatic signature. This started out just copying FR since you asked me to.

Reply:
Overall, though, `KeyVaultSettingValue` is an implementation detail. You wanted a `Value` property, but it's not otherwise constructable. Everything is done through `KeyVaultSetting` otherwise, so that we can control the inputs for what the service supports currently. In hindsight, this was probably a bad service design: it's (intrinsic) polymorphism via string serialization.